### PR TITLE
Support for adding stubs to an existing imposter

### PR DESCRIPTION
**Pull request by SShatun**
HI! Made fixes for previous pull request https://github.com/brunns/mbtest/pull/34

### CODE REVIEW

File: src/mbtest/imposters/stubs.py [hunk 1]
```diff
@@ -47,3 +47,39 @@ def from_structure(structure: JsonStructure) -> "Stub":
             [Predicate.from_structure(predicate) for predicate in structure.get("predicates", ())],
             responses,
         )
+
+
+class AddStub(JsonSerializable):
```
Comment (brunns) [inline src/mbtest/imposters/stubs.py:52]: Any reason we can't we use the standard Stub class here?

File: src/mbtest/server.py [hunk 2]
```diff
@@ -152,6 +153,26 @@ def add_imposters(self, definition: Union[Imposter, Iterable[Imposter]]) -> None
             definition.attach(self.host, post.json()["port"], self.server_url)
             self._running_imposters.append(definition)
 
+    def add_stubs(self, definition: Union[Stub, Iterable[Stub]], port: int):
```
Comment (brunns) [inline src/mbtest/server.py:156]: A stub should be added to an imposter rather than a server, no?

File: src/mbtest/imposters/stubs.py (hunk 1 above)
Comment (SShatun) [reply src/mbtest/imposters/stubs.py:52]: AddStub request looks like:
`{
  "index": 0,
  "stub":<your Stub>
}`

Of course you can write just like that ` json = {"stub": stub.as_structure()}` in `server.add_stub` 
but I tried  to keep project style.

File: src/mbtest/server.py (hunk 2 above)
Comment (SShatun) [reply src/mbtest/server.py:156]: I followed this logic: 
One port - one imposter. If you know the port you can identify an imposter.
If an imposter exists - add a stub to this imposter. 
If there no imposter on the port, then create a new imposter and add your stub to it.

Comment (SShatun) [reply src/mbtest/server.py:156]: Or you asked, why this method in Server class?
Yes, that makes the point. I will move it to Imposter

Comment (brunns) [reply src/mbtest/server.py:156]: Yup - I'd rather it was there. The impostor knows what its port is. And I'd rather adding a new impostor was explicit.

Comment (SShatun) [reply src/mbtest/server.py:156]: Surry, I made a new pull-request by mistake https://github.com/brunns/mbtest/pull/36
Moved the method to Imposter.

File: src/mbtest/imposters/stubs.py (hunk 1 above)
Comment (SShatun) [reply src/mbtest/imposters/stubs.py:52]: @brunns Do you think it is better to remove this class?

Comment (brunns) [reply src/mbtest/imposters/stubs.py:52]: We could just have `index` as an optional argument in the `add_stub()` method, perhaps?

Comment (SShatun) [reply src/mbtest/imposters/stubs.py:52]: Like that?
`    def add_stubs(self, definition: Union[Stub, Iterable[Stub]], index: int = None):
        if isinstance(definition, abc.Iterable):
            for stub in definition:
                self.add_stubs(stub)
        else:
            json = {"stub": definition.as_structure()}
            if index is not None:
                json["index"] = index`

Comment (SShatun) [reply src/mbtest/imposters/stubs.py:52]: Like that?
```    def add_stubs(self, definition: Union[Stub, Iterable[Stub]], index: int = None):
        if isinstance(definition, abc.Iterable):
            for stub in definition:
                self.add_stubs(stub)
        else:
            json = {"stub": definition.as_structure()}
            if index is not None:
                json["index"] = index```

Comment (SShatun) [reply src/mbtest/imposters/stubs.py:52]: Or 
`json = AddStub(stub=definition, index=index).as_structure()`

Comment (SShatun) [reply src/mbtest/imposters/stubs.py:52]: Made second variant in  #36

**Pull request closed without merging**